route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -133,39 +142,59 @@ namespace route_following_plugin
             if(end_dist < current_progress){
                 break;
             }
+            
+            auto following_lanelets = wm_->getRoute()->followingRelations(shortest_path[last_lanelet_index]);
 
-            resp.new_plan.maneuvers.push_back(
+            //if not already on last lanelet in path, check relation with next lanelet- follow lane change procedure if req, else lane follow
+            if(last_lanelet_index!= (shortest_path.size()-1) && identifyLaneChange(following_lanelets, shortest_path[last_lanelet_index + 1].id()))
+            {
+                //calculate constant start distance for lane change
+                double longl_travel_dist = (target_speed*(LANE_CHANGE_TIME_MAX + buffer_lanechange_time_)); //*config_param for time
+                double lane_change_start_dist;               
+                //lane change start distance needs to be constant                
```

Comment:
I don't understand this comment, we update it every timestep just a few lines down? Looks like we could also just directly initialize it to current_progress on 153 instead of splitting definition and initialization across 2 lines.

Reply:
This comment got left behind from when lane following was being called before lane change. I'll remove it.